include/nbl/builtin/hlsl/workgroup/fft.hlsl

```diff
@@ -93,15 +93,19 @@ struct exchangeValues<SharedMemoryAdaptor, float64_t>
     }
 };
 
+// Get the required size (in number of uint32_t elements) of the workgroup shared memory array needed for the FFT
+template <typename scalar_t, uint32_t WorkgroupSize>
+NBL_CONSTEXPR uint32_t sharedMemSize = 2 * WorkgroupSize * (sizeof(scalar_t) / sizeof(uint32_t));
```

Comment:
what namespace are you in?

Reply:
`workgroup::fft`, why?

Reply:
okay I see this mehtod should be user-facing while exchange values in the same namespace should not be user-facing. Where should I put this instead?

Reply:
okay if you want to make a template variable (before this wasn't possible, which is why workgroup scan stuff is structs) thats fine, can be in `workgroup::fft` namespace

Then have it as it is `SharedMemorySize` and make sure to report in bytes, or call it `SharedMemoryDWORDs` if you want to count in multiples of `uint32_t`

Furthermore `sizeof(complex<scalar_t>)` is better than `2*sizeof(scalar_t)` for understanding a readability.